[eas-cli] Remove Amplitude

kgc00 opened the pull request:
# Checklist
- [x] I've added an entry to CHANGELOG.md if necessary.
- [ ] I've tagged the changelog entry with [EAS BUILD API] if it's a part of a breaking change to EAS Build API (only possible when updating @expo/eas-build-job package).
 
# Why

This is a follow up to #563 which removes amplitude (currently we're double tracking to ensure event volumes match). Everything's been looking good, so it seems as good a time as any to remove amplitude.

# How

Deleted references to Amplitude, uninstalled packages.

As per comments in the previous PR, I renamed user settings with `amplitude` in the name to `analytics` and included some logic to check for both strings. Eventually, we can just check for the presence of `analytics...`

# Test Plan

Ran `yarn test`, tried enabling/disabling analytics, made sure events made it to the backend when they should.

dsokal commented on packages/eas-cli/src/analytics.ts:
```diff
@@ -20,40 +18,35 @@ const PLATFORM_TO_ANALYTICS_PLATFORM: { [platform: string]: string } = {
+  traits: Record<string, any>;
 } | null = null;
 
 export async function initAsync(): Promise<void> {
```
This block is the only thing we need to support old settings format:
```suggestion
export async function initAsync(): Promise<void> {
  // TODO: remove after some time
  const amplitudeEnabled = await UserSettings.getAsync('amplitudeEnabled', null);
  if (amplitudeEnabled !== null) {
    await UserSettings.setAsync('analyticsEnabled', amplitudeEnabled);
    await UserSettings.deleteKeyAsync('amplitudeEnabled');
  }
  const amplitudeDeviceId = await UserSettings.getAsync('amplitudeDeviceId', null);
  if (amplitudeDeviceId !== null) {
    await UserSettings.setAsync('analyticsDeviceId', amplitudeDeviceId);
    await UserSettings.deleteKeyAsync('amplitudeDeviceId');
  } 
  // TODO: cut here
```

Then, in the rest of code, you can rely only on the `analyticsEnabled` and `analyticsDeviceId` settings.

dsokal commented on packages/eas-cli/src/commands/analytics.ts:
```diff
@@ -12,13 +12,15 @@ export default class AnalyticsView extends EasCommand {
-      await UserSettings.setAsync('amplitudeEnabled', status === 'on');
+      await UserSettings.setAsync('analyticsEnabled', status === 'on');
       Log.withTick(`${status === 'on' ? 'Enabling' : 'Disabling'} analytics.`);
     } else {
```
You can extract the logic from my previous comment to a function and call it here as well.

dsokal reviewed (changes requested):
I think this doesn't work as it should.

Consider the following scenario:
1. There is a user who disabled analytics with some old CLI version. This means they have `amplitudeEnabled` set to false.
2. In your PR, you're only checking the value of the legacy setting (`amplitudeEnabled`). There isn't any logic that changes the already set setting. I mean, if there is `{ "amplitudeEnabled": false }` in `user-settings.json` then it should be rewritten as `{ "analyticsEnabled": false }`.
3. This means that if we dropped the fallback to `amplitudeEnabled` in a future CLI version, the users who had disabled analytics would have them enabled back again.

I showed a possible implementation in my review comments. The approach I proposed should reduce the complexity of the PR.

----

For your future PRs to `eas-cli`, could you please push branches directly to this repository? 
- We have a CI workflow that checks the size difference of `eas-cli` with the applied changes. E.g. https://github.com/expo/eas-cli/pull/621/checks?check_run_id=3626805987, if the branch comes from another repository, it seems it can't add a comment to the PR.
- I'm not entirely sure if this is also related but we also have a PR template. It either doesn't show up for PR from forked repos or you didn't use it. I'm pointing this out because there is a checklist that reminds us to update the CHANGELOG. Having said that, could you please add a changelog entry?

kgc00 commented:
> I think this doesn't work as it should.
> 
> Consider the following scenario:
> 
> 1. There is a user who disabled analytics with some old CLI version. This means they have `amplitudeEnabled` set to false.
> 2. In your PR, you're only checking the value of the legacy setting (`amplitudeEnabled`). There isn't any logic that changes the already set setting. I mean, if there is `{ "amplitudeEnabled": false }` in `user-settings.json` then it should be rewritten as `{ "analyticsEnabled": false }`.
> 3. This means that if we dropped the fallback to `amplitudeEnabled` in a future CLI version, the users who had disabled analytics would have them enabled back again.
> 
> I showed a possible implementation in my review comment. The approach I proposed should reduce the complexity of the PR.
> 
> For your future PRs to `eas-cli`, could you please push branches directly to this repository?
> 
> * We have a CI workflow that checks the size difference of `eas-cli` with the applied changes. E.g. https://github.com/expo/eas-cli/pull/621/checks?check_run_id=3626805987, if the branch comes from another repository, it seems it can't add a comment to the PR.
> * I'm not entirely sure if this is also related but we also have a PR template. It either doesn't show up for PR from forked repos or you didn't use it. I'm pointing this out because there is a checklist that reminds us to update the CHANGELOG. Having said that, could you please add a changelog entry?

hey, thanks for the suggestion, Dominik. I 100% agree, and pulled it in. Also, sorry about working from my fork - I'll work from this repo in the future. (I also added in the checklist portion of the PR description)

dsokal commented on packages/eas-cli/src/commands/analytics.ts:
```diff
@@ -12,13 +12,15 @@ export default class AnalyticsView extends EasCommand {
       Log.withTick(`${status === 'on' ? 'Enabling' : 'Disabling'} analytics.`);
     } else {
-      const amplitudeEnabled = await UserSettings.getAsync('amplitudeEnabled', null);
+      const amplitudeEnabled = (await UserSettings.getAsync('amplitudeEnabled', null)) !== false;
```
```suggestion
```

We don't need to check for `amplitudeEnabled` anymore. This class extends `EasCommand`.

kgc00 replied:
😅  good catch!

dsokal reviewed (approved):
Just one more comment before merge 👏
